src/adapter/src/coord/sequencer/inner/cluster.rs

```diff
@@ -214,18 +232,45 @@ impl Coordinator {
 
         match (&config.variant, new_config.variant) {
             (Managed(config), Managed(new_config)) => {
-                self.sequence_alter_cluster_managed_to_managed(
-                    Some(session),
-                    cluster_id,
-                    config,
-                    new_config,
-                    ReplicaCreateDropReason::Manual,
-                )
-                .await?;
+                let needs_finalization = self
+                    .sequence_alter_cluster_managed_to_managed(
+                        Some(session),
+                        cluster_id,
+                        config,
+                        new_config.clone(),
+                        ReplicaCreateDropReason::Manual,
+                        strategy.clone(),
+                    )
+                    .await?;
+                if needs_finalization {
+                    return match strategy.condition {
+                        AlterClusterStrategyCondition::None => Ok(StageResult::Response(
+                            ExecuteResponse::AlteredObject(ObjectType::Cluster),
+                        )),
+                        AlterClusterStrategyCondition::For(duration) => Ok(StageResult::Handle(
+                            mz_ore::task::spawn(|| "Finalize Alter Cluster", async move {
+                                tokio::time::sleep(duration).await;
+                                let stage = ClusterStage::Finalize(FinalizeAlterCluster {
+                                    validity,
+                                    plan: plan.clone(),
```

Comment:
```suggestion
                                    plan,
```

This shouldn't need to clone.